src/dev/ci_setup/setup.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -e
+
+
+###
+### determine directory of kibana project
+###
+dir="$(pwd)"
+if [ -f "$dir/package.json" ] && [ -f "$dir/.node-version" ]; then
+  echo "Setting up node.js and npm in $dir"
+else
+  echo "src/dev/ci_setup/setup.sh must be run within a kibana repo"
+  exit 1
+fi
+
+
+###
+### download node
+###
+nodeVersion="$(cat $dir/.node-version)"
+nodeUrl="https://nodejs.org/download/release/v$nodeVersion/node-v$nodeVersion-linux-x64.tar.gz"
+nodeDir="${NODE_DIR:-"$dir/.node"}/$nodeVersion"
+echo " -- node: version=v${nodeVersion} dir=$nodeDir"
+
+echo " -- setting up node.js"
+if [ -x "$nodeDir/bin/node" ] && [ "$($nodeDir/bin/node --version)" == "v$nodeVersion" ]; then
+  echo " -- reusing node.js install"
+else
+  if [ -d "$nodeDir" ]; then
+    echo " -- clearing invalid node.js install"
```

Comment:
clearing *previous* node.js install

Reply:
The goal was to make this slightly more resilient to partial archive extractions by checking that node booted and reports the right version number, but previous isn't necessarily wrong. It will reuse previous installs though if they report compatibility